webserver/static/js/listens-viewer.js

```diff
@@ -17,18 +18,42 @@ var ListensViewer = React.createClass({
                           "in the data-user-id property.");
             return;
         }
-        $.get("/listen/user/" + container.dataset.userId, function(data) {
-            if (this.isMounted()) {
-                this.setState({
-                    userId: container.dataset.userId,
-                    listens: data.payload.listens
-                });
-            }
-        }.bind(this));
+        window.addEventListener('popstate', this.handleLoadData);
+        this.handleLoadData();
     },
     componentDidUpdate: function() {
         jQuery("abbr.timeago").timeago();
     },
+    handleLoadData: function () {
+        if (this.state.ignorePopState) { return; }
```

Comment:
This part is not working properly. http://chatlogs.metabrainz.org/brainzbot/metabrainz/2015-09-15/?msg=3351147&page=6